### PR TITLE
PAS-1362: Writing exchange rate json file

### PR DESCRIPTION
**Pull request by awasnik**

### CODE REVIEW

File: app/uk/gov/hmrc/currencyconversion/repositories/WriteExchangeRateRepository.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    val exchangeRateFormatData = Json.prettyPrint(jsValue)
+
+    val result = Try {
+      val writer = new PrintWriter(new File(pathToOutputFile))
+      writer.write(exchangeRateFormatData)
+      writer.close()
+    }
+    result match {
+      case Success(_) =>
+        Logger.info(" [XrsExchangeRateRequestWorker] Writing to file successful " + pathToOutputFile)
```
Comment (shinealex33) [inline app/uk/gov/hmrc/currencyconversion/repositories/WriteExchangeRateRepository.scala:47]: Please correct the class name in the Logger

File: app/uk/gov/hmrc/currencyconversion/repositories/WriteExchangeRateRepository.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  def deleteOlderFile: Unit = {
+    val sixMonthOldDate = LocalDate.now.minusMonths(6)
+    val deletePath = path + "-%02d".format(sixMonthOldDate.getMonthValue) +
+      sixMonthOldDate.getYear.toString.substring(2) + ".json"
+    val oldFilePath = new File(deletePath)
+    if(oldFilePath.exists()){
+      oldFilePath.delete()
+      Logger.info("[XrsExchangeRateRequestWorker] Six month old file deleted successfully " + oldFilePath)
```
Comment (shinealex33) [inline app/uk/gov/hmrc/currencyconversion/repositories/WriteExchangeRateRepository.scala:60]: Same as above - Please correct  the class name

File: conf/resources/json/exrates-monthly-0621.json [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    "exchangeRate" : 103.51,
+    "currencyName" : "Rupee"
+  }, {
+    "validFrom" : "2021-06-17",
+    "validTo" : "2021-06-17",
+    "currencyCode" : "JPY",
+    "exchangeRate" : 152.61,
+    "currencyName" : "Yen"
+  } ]
+}
```
Comment (shinealex33) [inline conf/resources/json/exrates-monthly-0621.json:63]: End of line please

File: test/resources/json/exrates-monthly-0621.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    "exchangeRate" : 103.51,
+    "currencyName" : "Rupee"
+  }, {
+    "validFrom" : "2021-06-17",
+    "validTo" : "2021-06-17",
+    "currencyCode" : "JPY",
+    "exchangeRate" : 152.61,
+    "currencyName" : "Yen"
+  } ]
+}
```
Comment (shinealex33) [inline test/resources/json/exrates-monthly-0621.json:63]: End of line please

**Review by shinealex33 — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/currencyconversion/repositories/WriteExchangeRateRepository.scala (hunk 1 above)
Comment (awasnik) [reply app/uk/gov/hmrc/currencyconversion/repositories/WriteExchangeRateRepository.scala]: done

File: conf/resources/json/exrates-monthly-0621.json (hunk 3 above)
Comment (awasnik) [reply conf/resources/json/exrates-monthly-0621.json:63]: as discussed, not needed as  this a generated file

File: test/resources/json/exrates-monthly-0621.json (hunk 4 above)
Comment (awasnik) [reply test/resources/json/exrates-monthly-0621.json:63]: same as above

File: app/uk/gov/hmrc/currencyconversion/repositories/WriteExchangeRateRepository.scala (hunk 2 above)
Comment (awasnik) [reply app/uk/gov/hmrc/currencyconversion/repositories/WriteExchangeRateRepository.scala]: done

File: test/uk/gov/hmrc/currencyconversion/repositories/ExchangeRateRepositorySpec.scala [hunk 5]
```diff
@@ -16,23 +16,74 @@
...
-        val fileNames = new File(getClass.getResource("/resources/xml/").toURI).list().toList
+        val fileNames = new File(getClass.getResource("/resources/json/").toURI).list().toList
 
         forAll(fileNames) { fileName =>
-          fileName should fullyMatch regex """exrates-monthly-\d{4}.xml"""
+          fileName should fullyMatch regex """exrates-monthly-\d{4}.json"""
         }
       }
+
+      "be written correctly" in {
```
[the remark this answers is not in the document]
Comment (awasnik) [reply test/uk/gov/hmrc/currencyconversion/repositories/ExchangeRateRepositorySpec.scala:52]: as discussed this is correct

**Review by shinealex33 — APPROVED** [review]
Approved

**Pull request merged**